### DISCUSSION TITLE
Ken Burns effect?

### DISCUSSION THREAD

**Discussion by semiligneous** [damongolding/immich-kiosk#73] [Ideas]
Is there any appetite for a [Ken Burns effect](https://en.wikipedia.org/wiki/Ken_Burns_effect) (panning/zooming across the images), I see only a passing mention to "Ken Burns" in https://github.com/damongolding/immich-kiosk/discussions/67.

Thanks!

**Comment by damongolding** [comment]
I've had three people either ask or mention it in passing. I would say I have an appetite to see if I could pull it off with CSS!

**Comment by damongolding** [comment]
Just spitballing here but would it (in your mind) be an additional transition or a config option in itself?

**Reply by semiligneous** [reply]
I think it would be an additional (new) config option (ken_burns=true, or ken_burns_speed=fast or whatever other type of options may exist), you would still want to dictate which transition to use _between_ images using the same transition switch as today.

**Reply by semiligneous** [reply]
the suggestion in https://github.com/damongolding/immich-kiosk/discussions/67 of utilising the face position data might be a cool additional feature - ken_burns_use_face_position=true - but that seems highly likely to be more effort!

**Reply by damongolding** [reply]
A fair bit more yeah. I did something similar before but it was a zoom in/out effect rather then Ken Burns. But it did you face boxes to pin point the middle point which would show all faces.

**Reply by semiligneous** [reply]
Yeah any type of dynamic movement, zooming in/out, or panning across would add a nice element to the slideshow. 

I don't know how those types of things are generally programmed into other slideshows/screensavers - is it just a random effect focussed on a random part of the photo, or is there more 'intelligence' that picks the focal points and movement?! I guess that's where the complexity might come in and become a large piece of work!

**Reply by jschwalbe** [reply]
> A fair bit more yeah. I did something similar before but it was a zoom in/out effect rather then Ken Burns. But it did you face boxes to pin point the middle point which would show all faces.

I made #67 - Not sure how you did it, but the calculation in my mind would be 4 parts, corresponding to the minimum rectangle that keeps all the faces in (no midpoint necessary). 
  - Top left would be like this: of all faces, find the smallest X coordinate and smallest Y coordinate = (X,Y)_top_left, 
  - Top right=(largest X, smallest Y), and so on.
Happy to move this discussion to #67 instead but this was getting more activity ;)

**Comment by damongolding** [comment]
It depends on the software. Most will be a random point but with AI being more prevalent they will tend to be “smart” I guess.

Zooming will be relatively easy (easier) to implement. .

**Comment by damongolding** [comment]
That is exactly how I did it then used the centre point of that box as the zoom point.

**Comment by johndoe0815** [comment]
Appreciate the ken burns effect, but it is not yet right. If you want to have the picture only slowly moving, you need to set a low image effect amount. But that does not only determine the speed, but also the amount of zoom, and both seem to be linked.
Even with a refresh of 10 sec. with a image effect amount of 110 or 120 the picture is only moving for the first about 5 seconds, then it stands still before the next picture comes in. With a higher image effect amount, the zoom is just faster. If I increase the refresh, the zoom becomes slower, so somehow it is calculated in a way that the zoom never covers the whole refresh period, so the picture is always standing still for the last 20% of time or so.
The image effect should generally cover the whole time window that a picture is displayed. Plus it should combine zoom and panning (horizontal/vertical movement). Both slow so that there is continuous movement. Kodi has a very nice implementation of this effect.

**Reply by damongolding** [reply]
What I think you want is a more linear variant of the Ken Burns effect. This implementation does zoom/pan for the entire duration of the refresh but it uses a custom [easing](https://medium.com/motion-in-interaction/animation-principles-in-ui-design-understanding-easing-bea05243fe3), which on slower/longer durations might seem like it's not moving. 

I don't use Kodi so I haven't seen its implementation. I will do some research when I'm back on this or if you have an example that would be great.

**Reply by johndoe0815** [reply]
I did not find a good video example of the kodi slideshow, but here in the first few seconds you can see the effect that I am looking for, a combination of zoom in or out and horizontal/vertical movement, going over the whole duration of the picture being displayed:
https://www.youtube.com/watch?v=e4OT072rKJk
Or here is another one:
https://www.youtube.com/watch?v=3OTv1AGwmYM

**Reply by damongolding** [reply]
> I did not find a good video example of the kodi slideshow, but here in the first few seconds you can see the effect that I am looking for, a combination of zoom in or out and horizontal/vertical movement, going over the whole duration of the picture being displayed:
> 
> https://www.youtube.com/watch?v=e4OT072rKJk
> 
> Or here is another one:
> 
> https://www.youtube.com/watch?v=3OTv1AGwmYM


Ahh yes, these are far more linear. The second example rotates between zooming and panning, it doesn't do both at the same time. Do you have a preference?

**Reply by johndoe0815** [reply]
As you openly ask: I checked again in Kodi, and it also does on or the other and cycles through, like in the second video but even slower. It is the combination of constant but changing movement, combined with a rather slow speed. At least for me one of the most important features of a slideshow, would be great if you can implement that!
